File: 28.find-the-index-of-the-first-occurrence-in-a-string.ts
/*
 * @lc app=leetcode id=28 lang=typescript
 *
 * [28] Find the Index of the First Occurrence in a String
 */

// @lc code=start
function strStr(haystack: string, needle: string): number {
  let index = -1

  if (haystack.indexOf(needle) === -1) {
    return index
  } else {
    index = haystack.indexOf(needle)
    return index
  }
};
// @lc code=end

File: go/28.find-the-index-of-the-first-occurrence-in-a-string.go
func strStr(haystack string, needle string) int {
    
    if len(needle) > len(haystack) {
        return -1
    }
    len_needle := len(needle)
    for i := 0 ; i< (len(haystack) - len_needle) + 1; i++ {

        subString  := haystack[i: i+ len_needle]

        if subString == needle {
            return i
        }

    }

    return -1

}